format/index.md

```diff
@@ -355,6 +360,10 @@ responses are called "compound documents".
 In a compound document, all included resources **MUST** be represented as an
 array of resource objects in a top-level `"included"` member.
 
+In a compound document, any resource which references other resources included 
+into the same document (either in top-level `"data"` or `"included"` members)
+**MUST** include references to those resources in its own `"links"` member.
+
```

Comment:
The spec says: 
> If a link object refers to resource objects included in the same compound document

But there is no requirement for the resource to include that "link object", which "refers to resource objects included in the same compound document". This is the position which this PR reflects: the spec does not wish to control `fields` and `include` separately.

Another way to say what I meant: if object A *actually* (in the entity-relationship space of the system behind the particular API) references object B, which happens to be returned in the same response in which A in returned, it is mandatory for the the object A to include B in its `links`. This sentence is the central point of this PR: objects are either in both `links` and `included`, or in neither of them. The other option, namely controlling `fields` and `include` independently, is discussed in #549.

I have added that "actually", although I realize that this wording is far from ideal.